### ISSUE TITLE
Value of loss function for find_MAP varies between backends for the same model

### ISSUE THREAD

**Issue by jessegrabowski** [pymc-devs/pymc-extras#580]
Take the following regression model for example:

```py
import pymc as pm
import numpy as np
import pymc_extras as pmx

X = np.random.normal(size=(100, 3))

with pm.Model() as m:
    beta = pm.Normal('beta', 0, 1, shape=(3,))
    alpha = pm.Normal('alpha')
    mu = alpha + X @ beta
    sigma = pm.Exponential('sigma', 1)
    y_hat = pm.Normal('y_hat', mu = mu, sigma=sigma)
    
    data = pm.sample_prior_predictive(draws=1).prior.y_hat.values.squeeze()
```

Fit with just one iteration so we can see the initial loss value:

```py
with pm.observe(m, {'y_hat':data}):
    idata = pmx.find_MAP(maxiter=1)
```
```                                                                                                    
  Minimizing                                    Elapsed   Iteration   Objective        ||grad||     
 ────────────────────────────────────────────────────────────────────────────────────────────────── 
  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0:00:00   3/3         27638.14142459   88.96384921  
                                                                                                    
```

```py
with pm.observe(m, {'y_hat':data}):
    idata = pmx.find_MAP(maxiter=1, compile_kwargs={'mode':'JAX'})
```

```
  Minimizing                                      Elapsed   Iteration   Objective      ||grad||     
 ────────────────────────────────────────────────────────────────────────────────────────────────── 
  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0:00:00   3/3         166.24722983   88.96384921  
```

They converge to the same answer, and the gradients are the same, but the loss values are very different, which is weird.

**Comment by Dekermanjian** [comment]
Hey @jessegrabowski, I was briefly looking into this and noticed that the difference between the JAX output and both Pytensor C backend and the NUMBA backend (which both match) is that the JAX loss value is not being squared. 

The JAX Array type is not being handled for in the `update_progressbar` method of the `ObjectiveWrapper` class in `better_optimize`.

On line 127 of `wrapper.py` if you add this then the objective will match:
```
        if isinstance(value, np.ndarray|jax.Array): # Here need to include jax.Array otherwise it is skipped
            value = (value**2).sum()
        elif isinstance(value, list | tuple):
            value = sum([x**2 for x in value])
```
